### ISSUE TITLE
Compile error with gcc 4.8.5 and centos7 64-bit and Xeon E5-2680 v2

### ISSUE THREAD

**Issue by zhonhel** [t0rakka/mango#61]
Hi, this library is awesome! But I encountered compile error. Could you please help me out ?

```
[***@*** build]# mkdir cmake
[***@*** build]# cd cmake
[***@*** cmake]# cmake3 -DENABLE_AVX512=ON -DBUILD_SHARED_LIBS=ON ..
-- The CXX compiler identification is GNU 4.8.5
-- The C compiler identification is GNU 4.8.5
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ - works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc - works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Compiler: GNU
-- Processor: x86_64
-- Build: Release
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- SIMD: AVX-512 (2015)
-- Configuring done
-- Generating done
-- Build files have been written to: ***/mango-master/build/cmake
[***@*** cmake]# make -j20
Scanning dependencies of target mango
[  0%] Building CXX object CMakeFiles/mango.dir/***/mango-master/source/mango/core/aes.cpp.o
[  1%] Building CXX object CMakeFiles/mango.dir/***/mango-master/source/mango/core/cpuinfo.cpp.o
[  1%] Building CXX object CMakeFiles/mango.dir/***/mango-master/source/mango/core/crc32.cpp.o
[  1%] Building CXX object CMakeFiles/mango.dir/***/mango-master/source/mango/core/compress.cpp.o
[  1%] Building CXX object CMakeFiles/mango.dir/***/mango-master/source/mango/core/buffer.cpp.o
[  2%] Building CXX object CMakeFiles/mango.dir/***/mango-master/source/mango/core/sha2.cpp.o
[  2%] Building CXX object CMakeFiles/mango.dir/***/mango-master/source/mango/core/string.cpp.o
c++: error: unrecognized command line option ‘-mavx512dq’
c++: error: unrecognized command line option ‘-mavx512vl’
c++: error: unrecognized command line option ‘-mavx512bw’
c++: error: unrecognized command line option ‘-mavx512dq’
c++: error: unrecognized command line option ‘-mavx512vl’
c++: error: unrecognized command line option ‘-mavx512bw’
make[2]: *** [CMakeFiles/mango.dir/***/mango-master/source/mango/core/cpuinfo.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [CMakeFiles/mango.dir/***/mango-master/source/mango/core/crc32.cpp.o] Error 1
c++: error: unrecognized command line option ‘-mavx512dq’
```

**Comment by t0rakka** [comment]
The compiler is too old and doesn't support AVX-512 instructions. The AVX-512 support was added in GCC 4.9 but I recommend at least GCC 8 as the support was improved over the years. The latest released version at this time is 10.1 or so and is pretty good. There were some AVX-512 bugs fixed in GCC 8 so that's why I recommend it at least.

**Comment by zhonhel** [comment]
Thanks a lot! 
I just now use gcc (GCC) 9.1.1 to compile. Compile completely with no error.
Thanks again~
